modules/src/tx_msg.rs

```diff
@@ -1,33 +1,30 @@
 use crate::ics24_host::error::ValidationError;
 use prost_types::Any;
-use tendermint::account::Id as AccountId;
 
 pub trait Msg: Clone {
     type ValidationError: std::error::Error;
+    type Raw: From<Self> + prost::Message;
 
-    // TODO -- clarify what is this function supposed to do & its connection to ICS26 routing mod.
+    // TODO: Clarify what is this function supposed to do & its connection to ICS26 routing mod.
     fn route(&self) -> String;
 
-    fn get_sign_bytes<M: From<Self> + prost::Message>(&self) -> Vec<u8> {
-        let mut buf = Vec::new();
-        let raw_msg: M = self.clone().into();
-        prost::Message::encode(&raw_msg, &mut buf).unwrap();
-        buf
-    }
-
     /// Unique type identifier for this message, to support encoding to/from `prost_types::Any`.
-    fn type_url(&self) -> String {
-        unimplemented!()
-    }
+    fn type_url(&self) -> String;
 
-    fn to_any<M: From<Self> + prost::Message>(&self) -> Any {
+    #[allow(clippy::wrong_self_convention)]
+    fn to_any(self) -> Any {
         Any {
             type_url: self.type_url(),
-            value: self.get_sign_bytes::<M>(),
+            value: self.get_sign_bytes(),
         }
     }
 
-    fn get_signers(&self) -> Vec<AccountId>;
+    fn get_sign_bytes(self) -> Vec<u8> {
+        let mut buf = Vec::new();
+        let raw_msg: Self::Raw = self.into();
```

Comment:
Hmm interesting that we're now avoiding a `clone` here. It used to be `let raw_msg: M = self.clone().into();`. Perhaps cloning is no longer necessary because we're not using generics anymore?

Later edit: I now see that we're consuming `self` in the call to `to_any` and `get_sign_bytes`, which explains the lack of cloning.